### ISSUE TITLE
New API proposal- Device management

### ISSUE THREAD

**Issue by maheshc01** [camaraproject/APIBacklog#50]
Related to PR #30 

Device management was identified as priority API as part of OGW drop 2.
The API will focus on managing device subscription state . Initial scope to include add, activate, suspend, restore, deactivate and delete.
Additional information is available as part of the PR.

**Comment by maheshc01** [comment]
Hi @TEF-RicardoSerr I havent been able to join the API backlog call.I will plan on joining the call on 27th of this month.
Are we waiting on additional supporting operators to take this API to TSC or anything else pending on this?

**Comment by gmuratk** [comment]
I wanted to ask a question based on what was presented in the API Backlog Meeting on October 24, 2024.  Doesn't the scope of the feature(s) proposed fall in to Operate API?

**Comment by maheshc01** [comment]
Hi @gmuratk, IoT has been recognized as a family of APIs for CAMARA to support under GSMA Opengateway and this proposal provides basic set of capabilities for IoT device mgmt. This API will still falls in the category of northbound APIs as is the case for all CAMARA APIs. One uniqueness i like to note for this family of APIs is that rather than ASPs being the predominant users as in the case for other CAMARA APIs thus far, this API would be more widely used by enterprise customers (small/medium/large) in my opinion.
Focus of Operate APIs is quite different to what is proposed here. Operate APIs cater to standardizing the developer engagement and experience in terms of registration and accessing the APIs, inventory of the APIs, monitoring the utilizations, billing etc  

Hope this helps. 
would like to hear back on your thoughts and other's opinion as well on this.

**Comment by gmuratk** [comment]
Hi @maheshc01,
I have no arguments against IoT being in the scope.  The reason I raised the question is mainly due to proposed capabilities.

**Comment by maheshc01** [comment]
[IoT Provisioning APIs.pdf](https://github.com/user-attachments/files/17563002/IoT.Provisioning.APIs.pdf)

@gmuratk The PR which has the slides i shared during the call is not approved yet. Reattaching it again here for reference.
the scope identified for the API in the proposal covers add, activate, suspend, restore, deactive and delete in the context of device mgmt.
Could you elaborate further on what you mean by these capabilities falling in the scope of operate APIs?

**Comment by gmuratk** [comment]
@maheshc01 thanks for sharing the slides.  For example, capabilities for Adding devices and Activating service  **_may_** have an overlap with `service provisioning` that is listed as part of 'Operate API' scope in [CAMARA Project Charter](ProjectCharter.md).

**Comment by eric-murray** [comment]
@chenfr2-ChinaTelecom
Can you discuss possibly merging the scopes of #95 and #96 with this API proposal here in this issue? Thanks.

**Comment by chenfr2-ChinaTelecom** [comment]
Hi , @maheshc01,
We proposed [#95](https://github.com/camaraproject/APIBacklog/issues/95) (IoT SIM Status Mgmt) which is not only the management ability of device lifecycle management but also communication functions, and [#96](https://github.com/camaraproject/APIBacklog/issues/96) (IoT SIM Fraud Prevention) which is mainly about the management of device-SIM card binding and regional restriction functions.

We had explored your proposal about Device Management and suppose that it is mainly focused on the lifecycle management of IoT SIM cards, which might be a small part of the device management in IoT, the scope of this proposal might covers a much wider range than your actual API capabilities.

So we suggest that perhaps you could change your proposal name to "Device LifeCycle Management"? Or our 3 proposals could form an new API family called “Device management”， and yours could be named as "Device Management - LifeCycle Mgmt", and our two proprosals could be named as "Device Management -- Communication Function Mgmt".

**Comment by maheshc01** [comment]
Hi @chenfr2-ChinaTelecom , as discussed in the API backlog call, I have reviewed your proposal for IoT status mgmt API. The proposal has scope to activation and deactivation of various services like voice, SMS, data and specific APNs. 
In my proposal of device mgmt we do have activate, deactive, resume type if mgmt actions in scope. and being IoT device we are more focussed on data only.  But i think we can combine there 2 APIs into 1 yaml and we can get into more details during the yaml specification. 
Also as discussed, IoT sim fraud prevention will need to be part of a different yaml. There are also existing sim fraud detection APIs in Camara which can be applied to IoT devices. you will have to work with the API backlog group for next steps for it.

tagging @jgarciahospital for review and comments.

**Comment by jgarciahospital** [comment]
Thanks @maheshc01 , summarizing then:

- Device Management & IoT status mgmt APIs can be merged in the same specification track (AP templates to be modified accordingly, please decide in which of the current PRs will be included (suggestion is to do it in DeviceManagement one)
- IoT SIM Fraud still pending, not related to Device Management but with other current CAMARA APIs, to be discussed next backlog meeting @chenfr2-ChinaTelecom please confirm meeting RSVP

**Comment by chenfr2-ChinaTelecom** [comment]
Hi @maheshc01 
As discussed in the last meeting at Feb 13th, IoT status mgmt APIs will be merged into Device Management, could you please modify your AP templates to include our IoT status mgmt API ? Thanks

Hi @jgarciahospital 
should we also modify the "API family name" field value from "IoT SIM Status Mgmt" to "Device Management" accordingly in our IoT status mgmt APIs templates, and what else should we do ?

**Comment by jgarciahospital** [comment]
Thanks @chenfr2-ChinaTelecom . I think you can directly modify #30 by including a new commit with the use cases from IoT SIM Status, and therefor both use cases will be directly handled under the same umbrella. On the other side, as the use case will be already covered here, #95 could be discarded. is that ok for you? (@maheshc01 cc)
